cgra/CGRAKingMeshRTL.py

```diff
@@ -149,11 +149,26 @@ def construct( s, DataType, PredicateType, CtrlType, width, height,
         s.tile[i].to_mem_waddr.rdy //= 0
         s.tile[i].to_mem_wdata.rdy //= 0
 
+  # verbose trace if verbosity > 0
+  def verbose_trace(s, verbosity=1):
+    res = ''
+    for (i, x) in enumerate(s.tile):
+      res += "# [tile" + str(i) + "]: " + x.trace(verbosity=verbosity) + x.ctrl_mem.trace(verbosity=verbosity) + '\n'
+    res += f"\ndata_mem: {s.data_mem.trace(verbosity=verbosity)}"
+    res += "------\n\n"
+    return res
+
   # Line trace
-  def line_trace( s ):
+  def line_trace(s):
     # str = "||".join([ x.element.line_trace() for x in s.tile ])
     # str += " :: [" + s.data_mem.line_trace() + "]"
-    res = "||\n".join([ (("[tile"+str(i)+"]: ") + x.line_trace() + x.ctrl_mem.line_trace())
-                              for (i,x) in enumerate(s.tile) ])
+    res = "||\n".join([(("[tile" + str(i) + "]: ") + x.line_trace() + x.ctrl_mem.line_trace())
+                       for (i, x) in enumerate(s.tile)])
     res += "\n :: [" + s.data_mem.line_trace() + "]    \n"
     return res
+
+  def trace(self, verbosity = 0):

```

Comment:
How about now, add a trace entrance, but in *_test.py, function name has to be line_trace(). @tancheng

Reply:
[This](https://github.com/tancheng/VectorCGRA/blob/49a40187f90a8d67b0a410c0d3bc5944422e22e6/cgra/test/CGRAKingMeshRTL_test.py#L61) has to be line_trace(), while [this](https://github.com/tancheng/VectorCGRA/blob/49a40187f90a8d67b0a410c0d3bc5944422e22e6/cgra/test/CGRAKingMeshRTL_test.py#L62) could just be trace(), right?

Reply:
Oops, ignore above, build error when PyMTL trys to generate .py, *_test.py has to call method line_trace() ... Let me revert commit tmr and change as you mentioned.